src/server/server.cpp

```diff
@@ -1625,7 +1632,29 @@ void server::handle_player_in_game(socket_ptr socket, std::shared_ptr<simple_wml
 
 		if(user_handler_) {
 			const simple_wml::node& m = *g.level().root().child("multiplayer");
-			user_handler_->db_insert_game_info(uuid_, g.db_id(), game_config::wesnoth_version.str(), g.name(), m["mp_scenario"].to_string(), m["mp_era"].to_string(), g.is_reload(), m["observer"].to_bool(), !m["private_replay"].to_bool(), g.has_password());
+
+			// [addon] info handling
+			std::string scenario_addon_id = "";
+			std::string scenario_addon_version = "";
+			std::string era_addon_id = "";
+			std::string era_addon_version = "";
+			for(const auto& addon : m.children("addon")) {
+				for(const auto& content : addon->children("content")) {
+					if(content->attr("type").to_string() == "scenario") {
+						scenario_addon_id = addon->attr("id").to_string();
+						scenario_addon_version = addon->attr("version").to_string();
+					} else if(content->attr("type").to_string() == "era") {
+						era_addon_id = addon->attr("id").to_string();
+						era_addon_version = addon->attr("version").to_string();
+					} else if(content->attr("type").to_string() == "modification") {
+						user_handler_->db_insert_modification_info(uuid_, g.db_id(), content->attr("id").to_string(), addon->attr("id").to_string(), addon->attr("version").to_string());
+					} else {
+						WRN_SERVER << "[multiplayer][addon][content] sent with unknown type.\n" << simple_wml::node_to_string(*addon) << '\n';
+					}
+				}
+			}
```

Comment:
For my understanding: `[multiplayer]` contains multiple `[addon]` tags that contain multiple` [content]` tags, right? There can only be one `[content] type=era` and `[content] type=scenario` or multiple? (If there can be multiple what does that mean?)

Reply:
There can only be at most one `[content] type=era` and one `[content] type=scenario` (having multiple wouldn't make any sense).  There can be zero or more `[multiplayer][addon]` tags (ie: a vanilla map with Default era and no modifications would have zero), and each `[multiplayer][addon]` can contain one or more [content] tags (ie: one add-on could contain a scenario, an era, and multiple modifications).  A couple examples are attached [here](https://github.com/wesnoth/wesnoth/pull/4884#issuecomment-660567192).

The scenario ID and era ID are still taken directly from the `[multiplayer] mp_scenario=` and `[multiplayer] mp_era=` since mainline content wouldn't appear in the `[multiplayer][addon]` tags seeing as those are not add-ons.

Reply:
Thanks for the explanation.